python/graphstorm/dataloading/dataloading.py

```diff
@@ -917,3 +921,170 @@ def __iter__(self):
 
     def __next__(self):
         return self.dataloader.__next__(), self.unlabeled_dataloader.__next__()
+
+####################### Distillation #############################
+
+class DataProvider:
+    r"""Data Provider. Combines a file sampler and a dataloader generator,
+    and streamingly provides an iterable over a dataset.
+
+    Parameters:
+    ----------
+        dataloader_generator : DataloaderGenerator: 
+            A dataloader generator
+            Generates dataloader based on given file path.
+        dataset_path str : 
+            Path to the data files.
+        shuffle : bool
+            Set to ``True`` to have the files reshuffled at every epoch (default: ``False``).
+        infinite : bool
+            Set to ``True`` to have the files sampled infinitely (default: ``False``).
+        local_rank : int
+            Local rank for the trainer (default: ``-1``).
+        is_train : bool
+            Set to ``True`` if the provider is for training set (default: ``True``).
+    """
+
+    def __init__(
+        self,
+        dataloader_generator,
+        dataset_path,
+        shuffle=False,
+        infinite=False,
+        local_rank=-1,
+        world_size=1,
+        is_train=True,
+    ):
+        super().__init__()
+        # TODO (HZ): implement prefetch to put files into self.dataset_future
+        self.dataset_future = []
```

Comment:
If this is not used in this PR, please remove it. You can keep the TODO.

Reply:
sure